tests/phpunit/tests/load/wpGetDevelopmentMode.php

```diff
@@ -43,4 +44,118 @@ public function test_wp_get_development_mode_filter_invalid_value() {
 		$_wp_tests_development_mode = 'invalid';
 		$this->assertSame( '', wp_get_development_mode() );
 	}
+
+	/**
+	 * Tests that `wp_in_development_mode()` returns expected results.
+	 *
+	 * @ticket 57487
+	 * @dataProvider data_wp_in_development_mode
+	 */
+	public function test_wp_in_development_mode( $current, $given, $expected ) {
+		global $_wp_tests_development_mode;
+
+		$_wp_tests_development_mode = $current;
+
+		if ( $expected ) {
+			$this->assertTrue( wp_in_development_mode( $given ) );
+		} else {
+			$this->assertFalse( wp_in_development_mode( $given ) );
+		}
+	}
+
+	/**
+	 * Data provider that returns test scenarios for the `test_wp_in_development_mode()` method.
+	 *
+	 * @return array Test scenarios, each one with 3 entries for currently set development mode (string), given
+	 *               development mode (string), and expected result (boolean).
```

Comment:
```suggestion
	 * @return array[]
```

This has become a faux standard while some discussions are ongoing about the best way to document dataproviders without having to double up the `@params` in the test.